calliope/config/annual-energy-balance.yaml

```diff
@@ -0,0 +1,17 @@
+constraints:
+  annual_carrier_prod_group_balance:
+    foreach: [nodes]
+    equation: "$annual_carrier_prod_group1 == $annual_carrier_prod_group2 * $factor2"
+    where: "supply_gas"
+    components:
+      annual_carrier_prod_group1:
+        - expression: sum(carrier_prod[techs=$tech_group_1], over=timesteps)
+      annual_carrier_prod_group2:
+        - expression: sum(carrier_prod[techs=$tech_group_2], over=timesteps)
+      factor2:
+        -expression: 3
```

Comment:
Yeah, since the constraint is per node it should be possible to just have a custom parameter defined at every node with "supply_gas" and that will end up in the model dataset. The only issue right now is it will raise a warning about a possible misspelling.